tools/environment/export.go

```diff
@@ -0,0 +1,450 @@
+	ex.writeObject("auth", "magiclink", m["magiclink"], ExtractOthers, StripModel)
+	ex.writeObject("auth", "enchantedlink", m["enchantedlink"], ExtractOthers, StripModel)
+	ex.writeObject("auth", "otp", m["otp"], ExtractOthers, StripModel)
+	ex.writeObject("auth", "totp", m["totp"], ExtractOthers, StripModel)
+	ex.writeObject("auth", "saml", m["saml"], ExtractOthers, StripModel)
+	ex.writeObject("auth", "oauth", m["oauth"], ExtractOthers, StripModel)
+	ex.writeObject("auth", "webauthn", m["webauthn"], ExtractOthers, StripModel)
+	ex.writeObject("auth", "password", m["password"], ExtractOthers, StripModel)
+	ex.writeObject("auth", "oidcidp", m["oidcidp"], ExtractOthers, StripModel)
+
+	fmt.Println("  - Flows")
+	ex.writeFlows(m["flows"])
+
+	fmt.Println("  - Theme")
+	ex.writeTheme(m["theme"])
+
+	if len(ex.AssetMappings) > 0 {
+		fmt.Println("  - Assets")
+		assets := map[string]any{
+			"mapping": ex.AssetMappings,
+		}
+		ex.writeObject("assets", "assets", assets, ExtractNone, StripNone)
+	}
+
+	fmt.Println("* Done")
+
+	return nil
+}
+
+func (ex *Exporter) printProject(m map[string]any) bool {
+	settings, ok := m["settings"].(map[string]any)
+	if !ok {
+		return false
+	}
+	projectName, ok := settings["projectName"].(string)
+	if !ok {
+		return false
+	}
+	version, ok := settings["version"].(float64)
+	if !ok {
+		return false
+	}
+	fmt.Printf("  - Project: %s\n", projectName)
+	fmt.Printf("  - Revision: %d\n", int(version))
+	return true
+}
+
+func (ex *Exporter) writeObject(dir, filename string, object any, extract ExtractMask, strip StripMask) error {
+	m, ok := object.(map[string]any)
+	if !ok {
+		return nil
+	}
+
+	if strip&StripVersion != 0 {
+		delete(m, "version")
+	}
+	if strip&StripID != 0 {
+		delete(m, "id")
+	}
+	if strip&StripModifiedTime != 0 {
+		delete(m, "modifiedTime")
+	}
+	if strip&StripProjectName != 0 {
+		delete(m, "projectName")
+	}
+
+	path, err := ex.ensurePath(dir)
+	if err != nil {
+		return err
+	}
+
+	fullpath := filepath.Join(path, filename+".json")
+
+	if ex.Extract&extract != 0 {
+		if err := ex.extractAssets(fullpath, m); err != nil {
+			return err
+		}
+	}
+
+	bytes, err := json.MarshalIndent(m, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	if err := os.WriteFile(fullpath, bytes, 0644); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (ex *Exporter) writeFlows(arr any) {
+	flowIds := []string{}
+	if flows, ok := arr.([]any); ok {
+		for i := range flows {
+			if object, ok := flows[i].(map[string]any); ok {
+				if flow, ok := object["flow"].(map[string]any); ok {
+					if flowId, ok := flow["id"].(string); ok {
+						flowIds = append(flowIds, flowId)
+						ex.writeFlow(flowId, object)
+					}
+				}
+			}
+		}
+	}
+	if len(flowIds) > 0 {
+		sort.Strings(flowIds)
+		list := map[string]any{"flows": flowIds}
+		ex.writeObject("flows", "flows", list, ExtractNone, StripNone)
+	}
+}
+
+func (ex *Exporter) writeFlow(flowId string, object map[string]any) {
+	ex.ensurePath("flows")
```

Comment:
do you want to get the error response here?

Reply:
I don't think it's critical. In general I need to do another pass on the error handling all over the code, perhaps just `os.Exit` in several places instead of returning `error` types.